File: Quarkus/jwt-outcome/src/test/java/local/example/outcome/OutcomeResourceTest.java
package local.example.outcome;

import io.netty.handler.codec.http.HttpResponseStatus;
import io.quarkus.test.junit.QuarkusTest;

import io.restassured.http.ContentType;

import org.junit.jupiter.api.Test;

import static io.restassured.RestAssured.given;

import static org.hamcrest.CoreMatchers.is;

@QuarkusTest
public class OutcomeResourceTest {

    private static final String JSON_STRING = "{\"country\":\"Fantasy\",\"city\":\"Someone\",\"name\":\"James Doe\",\"email\":\"james.doe@example.local\",\"subject\":\"some sample subject\"}";

    @Test
    public void feedbackOutcomeEndpointTest() {
        given()
          .when().get("/outcome")
          .then()
             .statusCode(200)
             .body(is("--- jwt-outcome feedback ---"));
    }

    @Test
    public void unsignedEndpointTest() {
        given()
                .contentType(ContentType.JSON)
                .body(JSON_STRING)
                .when().post("/outcome/unsigned")
                .then().statusCode(HttpResponseStatus.OK.code());
    }

    @Test
    public void signedEndpointTest() {
        // TODO
    }
}

File: Quarkus/REST/dummy-users/src/test/kotlin/local/example/outcome/OutcomeResourceTest.kt
package local.example.outcome

import io.quarkus.test.junit.QuarkusTest

import io.restassured.http.ContentType
import io.restassured.RestAssured.given

import org.apache.http.HttpStatus

import org.hamcrest.CoreMatchers.`is`
import org.junit.jupiter.api.Test

@QuarkusTest
class OutcomeResourceTest {

    @Test
    fun testReadAllEndpoint() {
        given()
          .`when`().get(BASE_PATH)
          .then().statusCode(HttpStatus.SC_OK)
    }

    @Test
    fun testCreateEndpoint() {
        given().contentType(ContentType.JSON)
            .body(JSON_DATA).`when`().post(BASE_PATH)
            .then().statusCode(HttpStatus.SC_OK)
            .body(`is`(JSON_OUTCOME))
    }

    companion object {
        private const val BASE_PATH = "/outcome"
        private const val JSON_DATA =
            "{\"login\":{\"uuid\":\"c066898d-828a-4af8-bc52-f3d1d06156d9\"},\"name\":{\"title\":\"\",\"first\":\"\",\"second\":\"\"},\"email\":\"\",\"phone\":\"\",\"cell\":\"\",\"picture\":{\"large\":\"\",\"medium\":\"\",\"thumbnail\":\"\"}}"
        private const val JSON_OUTCOME =
            "[{\"login\":{\"uuid\":\"c066898d-828a-4af8-bc52-f3d1d06156d9\"},\"name\":{\"title\":\"\",\"first\":\"\",\"second\":\"\"},\"email\":\"\",\"phone\":\"\",\"cell\":\"\",\"picture\":{\"large\":\"\",\"medium\":\"\",\"thumbnail\":\"\"}}]"
    }

}
